### PR TITLE
Allow modules/notification/slack.py to send message using user/bot tokens.

### PR DESCRIPTION
**Pull request by Rumbles**
##### SUMMARY
Add the ability to send slack messages using the more up to date bot/user tokens which are now supported.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
notification/slack.py

##### ADDITIONAL INFORMATION
The existing module only supports the webhook method of sending messages to slack, currently slack recommends using bot/user tokens instead, but the module didn't support these. This update adds support for these, but should not affect existing support for the older token types that were already supported.

### CODE REVIEW

File: lib/ansible/modules/notification/slack.py [hunk 1]
```diff
@@ -46,6 +46,8 @@
         be locked by your Slack admin and you must request access.  It is there
         that the incoming webhooks can be added.  The key is on the end of the
         URL given to you in that section.
+      - You can also use user or bot tokens, (U(https://api.slack.com/docs/token-types)),
```
Comment (resmo) [inline lib/ansible/modules/notification/slack.py]: would you mind add a `Beginning with 2.10, you can...` to indicate the new feature with 2.10

Comment (Rumbles) [reply lib/ansible/modules/notification/slack.py]: Updated :+1:

**Review by resmo — APPROVED** [review]
shipit

Thanks!

**Comment by michalmiddleton** [comment]
Hello, is there a way to add a bit to this commit? I was going to submit a PR that solves the core of this issue and adds a bit more usability as well.

IMO, there are 3 related improvements this module needs:
1. Support user/bot tokens, as does this PR. 
2. Correct input type for thread_id (as does https://github.com/ansible/ansible/pull/56862) 
3. Return thread_ts/thread_id when user/bot tokens are used, so they can be later reused

I foolishly coded all three changes and found these PRs right before submitting mine. What's the best course of actions going from here?

**Comment by sivel** [comment]
This module is now maintained at https://github.com/ansible-collections/community.general

This PR will have to be migrated for it to be considered for merge.

**Comment by gundalow** [comment]
https://github.com/ansible-collections/community.general/pull/123 has been merged, so this has been fixed in the `community.general` Ansible Collection.

**Pull request closed without merging**